File: java/src/main/java/edu/cmu/cs/cs214/rec02/points/PolarPoint.java
package edu.cmu.cs.cs214.rec02.points;

public class PolarPoint implements Point {
    private double len;
    private double angle;

    public PolarPoint (double len, double angle) {
        this.len = len;
        this.angle = angle;
    }

    public int computeX() {
        return (int) (this.len * Math.cos(this.angle));
    }

    public int computeY() {
        return (int) (this.len * Math.sin(this.angle));
    }

    public int getX () {
        return computeX();
    }

    public int getY () {
        return computeY();
    }
}

File: typescript/src/points/polarPoint.js
"use strict";

exports.__esModule = true;
exports.newPolarPoint = void 0;
function newPolarPoint(len, angle) {
    function computeX() {
        return Math.round(len * Math.cos(angle));
    }
    function computeY() {
        return Math.round(len * Math.sin(angle));
    }
    return {
        getX: function () {
            return computeX();
        },
        getY: function () {
            return computeY();
        }
    };
}
exports.newPolarPoint = newPolarPoint;
